### PR TITLE
Add `--use-libraries` option for `pod repo push`

### PR DESCRIPTION
**Pull request by neonichu**
See #3168, IMHO shouldn't get a dedicated changelog entry as it should've been included in that change.

### CODE REVIEW

File: lib/cocoapods/command/repo/push.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Push < Repo
 
         def self.options
           [['--allow-warnings', 'Allows pushing even if there are warnings'],
+           ['--use-libraries', 'Lint uses static libraries to install the spec'],
```
Comment (segiddins) [inline lib/cocoapods/command/repo/push.rb]: `Linter` ?

File: spec/functional/command/repo/push_spec.rb [hunk 2]
```diff
@@ -96,5 +96,36 @@ module Pod
       Dir.chdir(@upstream) { `git checkout master -q` }
       (@upstream + 'PushTest/1.4/PushTest.podspec').read.should.include('PushTest')
     end
+
+    def stub_validation
```
Comment (segiddins) [inline spec/functional/command/repo/push_spec.rb]: `before` ?

Comment (neonichu) [inline spec/functional/command/repo/push_spec.rb]: Yah, don't know what I was thinking :confused:

**Comment by segiddins** [comment]
:+1:

**Pull request merged**